test/test-exception.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+// Testcase to produce NodeReport on uncaught exception
+if (process.argv[2] === 'child') {
+  require('../').setEvents("exception");
+
+  function myException(request, response) {
+    throw new UserException('*** exception.js: testcase exception thrown from myException()');
+  }
+
+  function UserException(message) {
+    this.message = message;
+    this.name = "UserException";
+  }
+
+  myException(); 
+} else {
+  const tap = require('tap');
+  const common = require('./common.js');
+  const fs = require('fs');
+  const spawn = require('child_process').spawn;
+
+  var child = spawn(process.execPath, [__filename, 'child']);
+  child.on('exit', function(code, signal) {
+    tap.plan(3);
+    tap.notEqual(code, 0, 'Process should not exit cleanly');
```

Comment:
Okay, I'm going to make the test assert no exit code (null) and SIGILL signal as that is the current expected behaviour. We should then change it if we change the default nodereport behaviour.